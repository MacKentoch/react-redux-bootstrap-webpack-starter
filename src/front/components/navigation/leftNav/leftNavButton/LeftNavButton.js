// @flow

// #region imports
import React, { PureComponent } from 'react';
import { Link } from 'react-router-dom';
// #endregion

// #region flow types
type Props = {
  link: string,
  label: string,
  viewName: string,
  onClick: (event: SyntheticEvent<>, viewName: string) => any,

  ...any,
};

type State = {
  ...any,
};
// #endregion

class LeftNavButton extends PureComponent<Props, State> {
  // #region lifecycle
  render() {
    const { link, label } = this.props;
    return (
      <li>
        <Link to={link} onClick={this.handleLeftNavItemClick}>
          {label}
        </Link>
      </li>
    );
  }
  // #endregion

  handleLeftNavItemClick = (event: SyntheticEvent<>) => {
    const { onClick, viewName } = this.props;
    onClick(event, viewName);
  };
}

export default LeftNavButton;
