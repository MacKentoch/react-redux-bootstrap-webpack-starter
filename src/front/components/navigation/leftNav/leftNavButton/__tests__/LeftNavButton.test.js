// @flow

// #region imports
import React from 'react';
import renderer from 'react-test-renderer'; // needed both for snpashot testing but also to prevent errors from enzyme
import { MemoryRouter } from 'react-router';
import LeftNavButton from '../LeftNavButton';
// #endregion

// #region flow types
type Props = {
  link: string,
  label: string,
  viewName: string,
  onClick: (event: SyntheticEvent<>, viewName: string) => any,
};
// #endregion

describe('LeftNavButton component', () => {
  it('renders as expected', () => {
    const props: Props = {
      link: '1',
      label: 'view 1',
      viewName: 'view 1',
      onClick: () => {},
    };

    const component = renderer
      .create(
        <div>
          <MemoryRouter>
            <LeftNavButton {...props} />
          </MemoryRouter>
        </div>,
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});
