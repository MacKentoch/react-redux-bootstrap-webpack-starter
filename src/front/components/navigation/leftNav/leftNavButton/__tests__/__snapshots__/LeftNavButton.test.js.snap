// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LeftNavButton component renders as expected 1`] = `
<div>
  <li>
    <a
      href="1"
      onClick={[Function]}
    >
      view 1
    </a>
  </li>
</div>
`;
